java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIntOperations.java

```diff
@@ -99,6 +99,18 @@ public static IntToken abs(IntToken term, TermContext context) {
         return IntToken.of(term.bigIntegerValue().abs());
     }
 
+    public static IntToken log2(IntToken term, TermContext context) {
```

Comment:
Code seems legitimate to me.  However, why wouldn't `bitLength()` work here?

Reply:
bitLength ceilings the log2, which is not the behavior we want, we want it floored.